homeassistant/components/api.py

```diff
@@ -71,6 +72,64 @@ def get(self, request):
         return self.json_message('API running.')
 
 
+class APILoggerView(HomeAssistantView):
+    """View to handle logger settings."""
+
+    url = "/api/logger"
+    name = "api:logger"
+
+    # NOTE, this is a dupe from components.logger
+    LOGSEVERITY = {
+        50: 'CRITICAL',
+        50: 'FATAL',
+        40: 'ERROR',
+        30: 'WARNING',
+        30: 'WARN',
+        20: 'INFO',
+        10: 'DEBUG',
+        0: 'NOTSET'
+    }
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Return all the available loggers as a JSON object."""
+        loggers = {k: APILoggerView.LOGSEVERITY[v.getEffectiveLevel()]
+                   for k, v in logging.Logger.manager.loggerDict.items()
+                   if isinstance(v, logging.Logger)}
+        return self.json(loggers)
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Update the given loggers with accompanying log levels."""
+        hass = request.app['hass']
+        try:
+            data = yield from request.json()
+        except ValueError:
+            return self.json_message('Invalid JSON specified',
+                                     HTTP_BAD_REQUEST)
+
+        current_loggers = logging.Logger.manager.loggerDict
+        loggers_to_update = {}
+        for to_change, new_level in data.items():
+            if new_level not in APILoggerView.LOGSEVERITY.values():
+                return self.json_message('Invalid log level: %s', HTTP_BAD_REQUEST)
```

Comment:
Haven't used aiohttp's  server capabilities yet, so I can't comment on your questions. But I think you forgot to specify a value for `%s` here (also: I *personally* prefer `format` over `%`, but that's just a suggestion)

```python
return self.json_message("Invalid log level: {}".format(new_level), HTTP_BAD_REQUEST)
```

Reply:
Fixed now, thanks! I really don't like the format syntax as it feels way too verbose.